kernel/nano_core/build.rs

```diff
@@ -76,7 +82,7 @@ fn emit_built_rs_file() {
 
             custom_cfgs_str.push_str(&key);
             if !v.is_empty() {
-                custom_cfgs_str.push_str(&format!("=\"{}\"", v));
+                write!(custom_cfgs_str, "=\"{}\"", v).expect("Failed to add String value to custom_cfgs_str variable");
```

Comment:
how is this better? seems worse to me. What lint is suggesting this, so i can read about its rationale?

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#redundant_field_names

Reply:
The way I put it now helps avoid unnecessary heap allocation, and also adds an error mechanism to spot a bug early in case there's a bug, instead of ignoring it

Reply:
oh interesting, sounds good then. One suggestion:
```suggestion
                write!(custom_cfgs_str, "=\"{}\"", v).expect("Failed to write custom_cfgs_str");
```